server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java

```diff
@@ -61,31 +68,101 @@
     }
 
```

Comment:
I ported all tests which I were able to port which seemed relevant. `testPeerRecoveryForClosedIndices` won't work, since closed tables/indices are not part of sys.shards. `testDoNotCancelRecoveryForBrokenNode` turned out to be flaky, therefore will postpone it.

Reply:
Table information is stored inside `information_schema.tables`, such as `information_schema.tables.closed`.

Reply:
`seq_no_stats` and `retention_leases` are only available from `sys.shards` and therefore only available for non-closed tables.

Reply:
See https://github.com/elastic/elasticsearch/blob/cec6678587ee09bf6470fccd773cc32ea8517bc9/server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java#L345

Reply:
Ah I see, so we may want to provide shard info also for closed tables? I think this would even not a real breaking change, grey area. @mfussenegger any opinion to that?

Reply:
I also think it should be safe to include closed tables in `sys.shards`, but we should double check if there are still any `sys.shards` based health calculations and if they'd break and if any of the other metrics that the admin-ui displays in the table view or similar would act up